chandra_aca/fov_plot.py

```diff
@@ -0,0 +1,309 @@
+               edgecolors='red',
+               linewidth=.5,
+               marker='o',
+               s=175)
+    ax.scatter(fid['yang'], fid['zang'],
+               facecolors='none',
+               edgecolors='red',
+               marker='+',
+               linewidth=.5,
+               s=175)
+
+
+def _plot_field_stars(ax, stars, quat, red_mag_lim=None, bad_stars=None):
+    """
+    Plot plot field stars in yang and zang on the supplied
+    axes object in place.
+
+    :param ax: matplotlib axes
+    :param stars: astropy.table compatible set of records of agasc entries of stars
+    :param quat: attitude quaternion as a Quat
+    :param red_mag_lim: faint limit
+    :param bad_stars: boolean mask of stars to be plotted in red
+    """
+    stars = Table(stars)
+
+    # Add star Y angle and Z angle in arcsec to the stars table
+    yagzags = (radec2yagzag(star['RA_PMCORR'], star['DEC_PMCORR'], quat)
+               for star in stars)
+    yagzags = Table(rows=[(y * 3600, z * 3600) for y, z in yagzags], names=['yang', 'zang'])
+    stars = hstack([stars, yagzags])
+    ok = np.ones(len(stars), dtype='bool')
+
+    if red_mag_lim:
+        nsigma = 3.0
+        mag_error_low_limit = 1.5
+        randerr = 0.26
+        caterr = stars['MAG_ACA_ERR'] / 100.
+        error = nsigma * np.sqrt(randerr*randerr + caterr*caterr)
+        error[error < mag_error_low_limit] = mag_error_low_limit
+        faint = ((stars['MAG_ACA'] >= red_mag_lim)
+                 & ((stars['MAG_ACA'] - error) < red_mag_lim))
+        too_dim_to_plot = ((stars['MAG_ACA'] >= red_mag_lim)
+                            & ((stars['MAG_ACA'] - error) >= red_mag_lim))
+        # All of the indices are on stars, so reindex without too_dim_to_plot
+        stars = stars[~too_dim_to_plot]
+        faint = faint[~too_dim_to_plot]
+        ok = ok[~too_dim_to_plot]
+        ok[faint] = False
+        if bad_stars is not None:
+            bad_stars = bad_stars[~too_dim_to_plot]
+    if bad_stars is not None:
+        ok[bad_stars] = False
+
+    size = symsize(stars['MAG_ACA'])
+    ax.scatter(stars[ok]['yang'], stars[ok]['zang'],
+               c='black', s=size[ok])
+    if red_mag_lim:
+        ax.scatter(stars[faint]['yang'], stars[faint]['zang'],
+                   c=FAINT_STAR_COLOR, s=size[faint], edgecolor='none',
+                   alpha=FAINT_STAR_ALPHA)
+    if bad_stars is not None:
+        ax.scatter(stars[bad_stars]['yang'], stars[bad_stars]['zang'],
+                   c=BAD_STAR_COLOR, s=size[bad_stars], edgecolor='none',
+                   alpha=BAD_STAR_ALPHA)
+
+
+def star_plot(catalog=None, attitude=None, stars=None, title=None,
+              red_mag_lim=None, quad_bound=True, grid=True, bad_stars=None):
+    """
+    Plot a catalog, a star field, or both in a matplotlib figure.
+    If supplying a star field, an attitude must also be supplied.
+
+    :param catalog: Records describing catalog.  Must be astropy table compatible.
```

Comment:
Somewhere in the docstrings, probably here, there needs to be an exact enumeration of the required fields in the catalog.